jina/drivers/rank/aggregate/__init__.py

```diff
@@ -69,30 +69,39 @@ def _insert_query_matches(
 
     @staticmethod
     def _group_by(match_idx, col_name):
-        # sort by ``col
-        _sorted_m = np.sort(match_idx, order=col_name)
-        _, _doc_counts = np.unique(_sorted_m[col_name], return_counts=True)
-        # group by ``col``
-        return np.split(_sorted_m, np.cumsum(_doc_counts))[:-1]
+        """
+        Create an list of numpy arrays with the same ``col_name`` in each position of the list
+
+        :param match_idx: Numpy array of Tuples with document id and score
+        :param col_name:  Column name in the structured numpy array of Tuples
 
+        :return: List of numpy arrays with the same ``doc_id`` in each position of the list
+        :rtype: np.ndarray.
+        """
+        _sorted_m = np.sort(match_idx, order=col_name)
+        list_numpy_arrays = []
+        current = _sorted_m[col_name][0]
+        prev_val = 0
+        n_elements = len(_sorted_m[col_name])
+        for i, val in enumerate(_sorted_m[col_name]):
+            if val != current:
+                list_numpy_arrays.append(_sorted_m[prev_val:i])
+                prev_val = i
+                current = val
+            if i == n_elements-1 and val == current:                
```

Comment:
can this be handled outside the for loop when u are sure u handled the last element?, what is the problem here?